lib/sensu/extensions/handlers/relay.rb

```diff
@@ -121,17 +121,21 @@ class Endpoint
     # space of a single loop tick.
     MAX_QUEUE_SIZE = 16384
 
-    attr_accessor :connection, :queue
+    attr_accessor :connection, :queue, :max_queue_size
 
-    def initialize(name, host, port)
+    def initialize(name, host, port, queue_size = nil)
```

Comment:
You can save the if nil check below by defaulting to MAX_QUEUE_SIZE instead of nil.